All Mobs as pets

Chris6ix opened the pull request:
### What is the purpose of this pull request?

Adds all the remaining Vanilla Mobs as pets to UC
Will try to complete for UC 3

#### Changes
- [x] Added Cod pet
- [x] Added Donkey pet
- [x] Added Mule pet
- [x] Added Pufferfish pet
- [x] Added Glow Squid pet (Disabled because of lack of movement)
- [x] Added Ocelot pet
- [x] Added Salmon pet
- [x] Added Skeleton Horse pet
- [x] Added Squid pet (Disabled because of lack of movement)
- [x] Added Strider pet
- [x] Added Tadpole pet
- [x] Added Tropical Fish pet
- [x] Added Turtle pet
- [x] Added Wandering Trader pet
- [x] Added Cave Spider pet
- [x] Added Dolphin pet
- [x] Added Spider pet
- [x] Added Zombified Piglin pet
- [x] Added Drowned pet
- [x] Added Elder Guardian pet
- [x] Added Endermite pet
- [x] Added Evoker pet
- [x] Added Ghast pet (Disabled because of lack of movement)
- [x] Added Guardian pet
- [x] Added Hoglin pet
- [x] Added Husk pet
- [x] Added Magma Cube pet (Disabled because of movement issues)
- [x] Added Phantom pet (Disabled because of lack of movement)
- [x] Added Piglin Brute pet
- [x] Added Pillager pet
- [x] Added Ravager pet
- [x] Added Shulker pet (Of course doesn't move besides teleport, but that is ok for this pet imo)
- [x] Added Stray pet
- [x] Added Vindicator pet
- [x] Added Witch pet
- [x] Added Wither Skeleton pet
- [x] Added Zoglin pet
- [x] Added Zombie Villager pet
- [x] Added Zombie Horse pet
- [x] Added Illusioner pet
#### Some improvements/fixes to do:
Important Fixes: 
- Prevent Player from transforming a Zombie Villager into a normal villager with a weakness potion and a golden apple
- Prevent Hoglin and Piglin Brute from turning into zombified versions in the overworld
- Fix Magma Cube damaging the player
- Prevent Elder Guardian from giving player mining fatigue

Minor Fixes/Improvements:
- Prevent Player from interacting with certain pets using their breeding items (Such as Pigs using a carrot, Horses using Wheat, Hoglins using a Crimson Fungus etc...) (*I think* this is just a minor issue as most pets are spawned as a baby version and thus are not able to breed, but still something that needs to be fixed)
- Reduce land speed for Wandering Trader, Evoker, Vindicator, Illusioner
- Increase water speed for Cod, Salmon, Tadpole, Tropical Fish, Turtle and Drowned
- Increase land speed for Skeleton Horse
- Maybe try to get the strider to feel like it's in lava all the time so that it looks better and moves faster
- https://github.com/datatags/UltraCosmetics/pull/22#issuecomment-1237515494 + comment below
- The Elder Guardian is a bit "buggy" when swimming after you (Weird movement, like he's not really willing to follow the player, but is forced to...he kind of tries to turn around all the time)
- Fix burning protection for Zombie, Skeleton etc. bugging (https://imgur.com/a/ELaX3OR)

datatags commented:
I think all the MobChip stuff is pretty much figured out now, so let me know as soon as you're done so I can merge it and test it out 🙂

Chris6ix commented:
Hey @datatags, I'm done with adding and testing the pets, so you can check it out now and merge it to test it with MobChip 🙂
I will create a new issue with all the comments I added into the PR description on top so that you can keep track of the stuff that needs to be done to the pets before they can get released.

I will also update #23 with all the new pets that I had to disable, if you manage to fix them with MobChip, just comment under that issue and I will update it.

I also already added the burning protections to the pets that need it as I already knew how to do that, but as I have noticed the burning protection is a bit bugged, so you would also need to check that out...I have left a video to show you the bug.

If you manage to fix all the pets in this PR that had to be disabled, then UC now has 40 new pets (+ the Vex that you already managed to fix) 😁

If you need help testing the pets again with MobChip or need help testing some of the fixes, just ask me on discord, if I manage to free some time, I will help you 😉

datatags commented:
Looks great, thanks so much! Also I think I figured out a better way to prevent burning, I'll do that when I have a chance.
